test-analytics/frameworks/swift-testing.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Configuring swift-testing
+description: a testing framework for pure Swift applications and libraries.
+layout:
+  title:
+    visible: true
+  description:
+    visible: false
+  tableOfContents:
+    visible: true
+  outline:
+    visible: true
+  pagination:
+    visible: true
+---
+
+Swift Testing is a testing framework for Apple platforms.
+
+## Enabling XML Output
+Swift testing has built in XUnit XML reporting support. When you run your tests from the command line add `--xunit-output report.xml`to output an XML report to a file. Ex:
+
+```swift
+import Testing
+
+@Test func helloworld() {
+    let greeting = "hello, world!"
+    #expect(greeting == "hello, worldz!")
+}
+```
+```shell
+swift test --xunit-output report.xml
+```
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+  <testsuite name="TestResults" errors="0" tests="1" failures="1" time="0.001664833">
+    <testcase classname="MyCLITests" name="helloworld()" time="0.000832083">
+      <failure message="Expectation failed: (greeting &#8594; &quot;hello, world!&quot;) == &quot;hello, worldz!&quot;" />
+    </testcase>
+  </testsuite>
+</testsuites>
+
+```
+
+
+## Test Suite Naming
+
+Once stable Swift Testing will use the name of a test function or suite as the `name` attribute  of the `testcase`by default. You can override the name using a string argument to the `@Test` and `@Suite` annotations. **Unfortunately these names are not currently propagated to the XML output**. Once fixed, you will be able to adjust the output like this:
+
+```swift
+import Testing
+
+@Suite("best tests evar") struct MyCoolTests {
+    @Test("rad test name") func helloworld() {
+        ...
+    }
+}
+```
+would produce
```

Comment:
Let's do a full sentence here:

`The above would produce:`

It looks a little silly and needs more visual weight to be easy to spot